### ISSUE TITLE
Godot doesn't respect `maxPerStageDescriptorSamplers` limit

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#50444]
### Godot version

4.0 96a69899d0909291888828205138bbf9def5e6ff

### System information

All

### Issue description

Godot due not respecting `maxPerStageDescriptorSamplers` limit(32), crashes when running on Lavapipe

https://gitlab.freedesktop.org/mesa/mesa/-/issues/3772#note_990084

### Steps to reproduce

Compile and install Mesa
Try to run Godot editor via Lavapipe

**Comment by BastiaanOlij** [comment]
Is this with the mobile renderer or the clustered one? Probably both at this time because I don't believe we have checks in place at the moment but we're trying to constraint the mobile renderer, the clustered one may not be able to be limited to 16.

I'll have to investigate further, @reduz is this the issue you talked about a little while ago where we had to restrict our samplers in the mobile renderer?

**Comment by reduz** [comment]
@qarmin I don't think this is the case, I checked again and we use only 13/14 samplers, no idea why so many are being reported.
@BastiaanOlij no, that is sampled images (textures) not samplers.

**Comment by qarmin** [comment]
I'm almost sure that this is Godot problem, because project with mobile renderer works(it crashes, but later with segfault, not assert as in this situation)

**Comment by reduz** [comment]
Mobile renderer uses the same amount of samplers but less textures so I still think this is a Lavapipe bug.
